core/src/main/scala/org/http4s/HttpVersion.scala

```diff
@@ -35,20 +35,21 @@ object HttpVersion {
       case "HTTP/1.1" => Either.right(`HTTP/1.1`)
       case "HTTP/1.0" => Either.right(`HTTP/1.0`)
       case _ =>
-        new Parser(s).HttpVersion.run()(Parser.DeliveryScheme.Either).leftMap { _ =>
-          ParseFailure("Invalid HTTP version", s"$s was not found to be a valid HTTP version")
+        parser.parseAll(s).leftMap { case e =>
+          ParseFailure("Invalid HTTP version", e.toString)
```

Comment:
Is there a general pretty printer for `Error`?  Is such a thing possible?  We didn't offer much detail in the parboiled implementation, and don't have to here.  Just exploring what's possible.

Reply:
There is not. The expectations are sorted but there isn't much pretty about it. I have some utilities for this in bosatsu I can pull out (mostly about pointing at the part of the string that failed). That would be good to add to cats-parse.